sdk/textanalytics/Azure.AI.TextAnalytics/src/NamedEntityType.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Gets the entity type inferred by the text analytics service's named entity recognition model. The list of available types is described at https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/named-entity-types.
+    /// </summary>
+    [JsonConverter(typeof(NamedEntityTypeJsonConverter))]
+    public readonly struct NamedEntityType
```

Comment:
Per [guidelines](https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-enums), it also needs to implement `IEquatable<T>`

Reply:
Fixed.